### ISSUE TITLE
The kurdish language rendering character issue. 

### ISSUE THREAD

**Issue by MuhamadHaydar** [DavBfr/dart_pdf#1408]
**Describe the bug**

The issue with some characters of the Kurdish language (ێ،ڵ), That it does not render them in the correct way.

**To Reproduce**
Code snippet to reproduce the behavior:
```dart
// your code
```

**Expected behavior**

**it does not integrate these characters in the correct way with others**

I load the custom Kurdish font from assets that work with both Arabic and Kurdish characters,
I use the latest version of pdf (3.10.4). 

**Screenshots**

**The current pdf rendering state:** 
![Screenshot 2023-06-11 111039](https://github.com/DavBfr/dart_pdf/assets/49116664/68a27579-399d-4178-b177-a958361efbbe)

**What is expected to render:**

![Screenshot 2023-06-11 111354](https://github.com/DavBfr/dart_pdf/assets/49116664/91a76a5f-680d-4ee5-876f-193ed8b9f6c5)

**Flutter Doctor**

[√] Flutter (Channel stable, 3.10.2, on Microsoft Windows [Version 10.0.22621.1778], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Build Tools 2019 16.11.25)
[√] Android Studio (version 2022.2)
[√] VS Code (version 1.76.2)
[√] Connected device (4 available)
[√] Network resources

```
```

**Desktop (please complete the following information):**
 - iOS
 - Android

**Comment by sosocom** [comment]
you should choose the correct font for your language.  This is my example for chinese. 
` pw.MultiPage(
        pageTheme: _buildTheme(
          pageFormat,
          await PdfGoogleFonts.notoSerifHKRegular(),
          await PdfGoogleFonts.notoSerifHKBold(),
          await PdfGoogleFonts.notoSerifHKLight(),
        ),`

**Comment by MuhamadHaydar** [comment]
@sosocom 

It did not solve the problem, I used some Google fonts that are working fine with these characters, but when the plugin renders it, it is not the same as it looked on the web preview of Google fonts.

**Comment by AriDevC** [comment]
See _https://github.com/DavBfr/dart_pdf/issues/1163#issuecomment-1346702736_

**Comment by MuhamadHaydar** [comment]
I have not idea how to solve my character rendering issue.

**Comment by arrowxpr** [comment]
You have to screenshot your text widget (or entire screen) and then feed it into the pdf plugin. It's very easy
https://pub.dev/packages/screenshot

**Comment by AkamBarznji** [comment]
> You have to screenshot your text widget (or entire screen) and then feed it into the pdf plugin. It's very easy https://pub.dev/packages/screenshot

Could you please let us know

**Comment by arrowxpr** [comment]
> > You have to screenshot your text widget (or entire screen) and then feed it into the pdf plugin. It's very easy https://pub.dev/packages/screenshot
> 
> Could you please let us know

I just made this [repo](https://github.com/arrowxpr/Flutter-Kurdish-Text-PDF) for the workaround. Hope it helps.

**Comment by NALAWALAMURTUZA** [comment]
@AkamBarznji @arrowxpr 

Will you please share the code snippet for screen shot rendering in flutter web?

Thank you in advance.

**Comment by hamreen** [comment]
I'm also encountering the same issue with Kurdish characters like "ێ" (U+06CE), "ڵ" (U+06B5), and "ە" (U+06D5) not rendering properly when generating PDFs. I’ve tried using multiple fonts such as NotoSansKurdish and Tahoma, which render correctly in the Flutter UI, Microsoft Word, and other environments. However, when rendered in the dart_pdf package, the characters are disjointed, and they do not concatenate properly.

Based on the discussion here, I see that a workaround involving taking a screenshot of the widget has been suggested. While this may be a temporary solution, it would be ideal to have a more robust fix directly within the PDF rendering process. Screenshots are not suitable for every scenario, especially when dealing with dynamic content, larger documents, or when using platforms like FlutterFlow, where implementing screenshot-based solutions is more challenging.

I’ve also tested this issue with syncfusion_flutter_pdf, but the problem persists across both packages. Given the complexity of handling Kurdish characters in different environments, are there any upcoming fixes or other recommendations that might help with this issue?

Additionally, Kurdish is spoken by over 30 million people across several countries, and accurate text rendering in digital documents is essential for both business and personal applications. Improving support for Kurdish characters would have a significant impact, especially for apps and services that cater to this large audience.

I also noticed that similar issues have been reported in other frameworks (e.g., [Stimulsoft](https://forum.stimulsoft.com/viewtopic.php?p=167272&hilit=Central+Kurdish+Language+Unicode+Issue#p167272) and [Aspose](https://forum.aspose.com/t/converting-excel-to-pdf-rendering-central-kurdish-language-unicode-issue/221745)), so it seems like a broader Unicode rendering issue.

**Comment by MuhamadHaydar** [comment]
@hamreen 
Have you solved the character's (ڵ،ێ) issue of kurdish language?

**Comment by hamreen** [comment]
Yes, i downloaded the package and fixed it myself. but the root cause of the problem which is with the dart_bidi package.

**Comment by MuhamadHaydar** [comment]
Can you provide more details of the solution? 
i would love to see if it's possible or if you could provide your contact information to catch up 
about it.

**Comment by hamreen** [comment]
@MuhamadHaydar 
Sorry I cannot, I've edited the package and now it contains vital credentials about my project.
I hope the bug be fixed so that it becomes available globally.

**Comment by hamreen** [comment]
@MuhamadHaydar 
Download or clone this(dart_pdf) package, do changes mentioned [here](https://github.com/DavBfr/dart_pdf/pull/1751), use [fontforge ](https://fontforge.org/en-US/) to add missing glyphs for a specific font, then use it. Vola.

It really took months for me to do it. be patient.

**Comment by hamreen** [comment]
I have solved the problem and published it as a new package called hamreen_printing.
For more details, you can watch the following video, which explains how to print Kurdish letters correctly using this new package:

👉 https://youtu.be/hMOIpvJgKiQ

**Comment by MuhamadHaydar** [comment]
Thank you so much for putting in so much effort to fix this problem.

It would be great if you could create a video about FontForge on how to add glyphs to a specific font to help us to add more fonts and make more fonts available.  

Thanks again for your efforts.
